generators/importanceSampler.cc

```diff
@@ -0,0 +1,423 @@
+		printErr << "less than two particles to sample. Four-momentum conservation leaves no d.o.f.. No sampling necessary. Aborting..." << std::endl;
+		throw;
+	}
+	const double mMin = _massAndTPrimePicker->massRange().first;
+	const double mMax = _massAndTPrimePicker->massRange().second;
+	if (mMin > mMax) {
+		printErr << "mMin = " << mMin << " > mMax = " << mMax << ". Aborting..." << std::endl;
+		throw;
+	}
+
+	// follow the convention of the nBodyPhaseSpace classes
+	double mMinPar = _masses[0];
+	for (size_t part = 1; part < _nPart; ++part) {
+		mMinPar += _masses[part];
+
+		std::stringstream nameM;
+		std::stringstream texM;
+		nameM << "m";
+		texM << "m_{";
+		for (size_t p = 1; p <= part+1; ++p) {
+			nameM << p;
+			texM << p;
+		}
+		texM << "}";
+		BCParameter param(nameM.str(), mMinPar, mMax, texM.str());
+		if (part == _nPart-1) {
+			if (mMin == mMax) {
+				// X mass is fixed to one value
+				param.Fix(mMax);
+			} else {
+				param.SetLimits(mMin, mMax);
+			}
+		}
+		AddParameter(param);
+
+		// sample cos(theta), because dOmega = sin(theta) dtheta dphi = dcos(theta) dphi
+		std::stringstream nameTheta;
+		std::stringstream texTheta;
+		nameTheta << "cosTheta" << part+1;
+		texTheta << "cos(#theta_{" << part+1 << "})";
+		AddParameter(nameTheta.str(), -1., 1., texTheta.str());
+
+		std::stringstream namePhi;
+		std::stringstream texPhi;
+		namePhi << "phi" << part+1;
+		texPhi << "#phi_{" << part+1 << "}";
+		AddParameter(namePhi.str(), 0., 2*M_PI, texPhi.str());
+	}
+	for (size_t part = 0; part < _nPart; ++part) {
+		for (size_t qart = 0; qart < part; ++qart) {
+			std::stringstream name;
+			std::stringstream tex;
+			name << "M" << qart+1 << part+1 << 2;
+			tex << "m^{2}_{" << qart+1 << part+1 << "}";
+			AddObservable(name.str(), 0., mMax*mMax, tex.str());
+		}
+	}
+
+	resetFuncInfo();
+	BCAux::SetStyle();
+}
+
+
+double
+rpwa::importanceSampler::LogAPrioriProbability(const std::vector<double>& parameters)
+{
+#ifdef _OPENMP
+#pragma omp atomic
+#endif
+	++(_funcCallInfo[LOGAPRIORIPROBABILITY].nmbCalls);
+	TStopwatch timerTot;
+	timerTot.Start();
+
+	rpwa::nBodyPhaseSpaceKinematics nBodyPhaseSpace;
+	if (not initializeNBodyPhaseSpace(nBodyPhaseSpace, parameters, false)) {
+		return -std::numeric_limits<double>::infinity();
+	}
+
+	double phaseSpace = nBodyPhaseSpace.calcWeight() / std::pow(parameters[3*_nPart-6] - _mSum, _nPart-2);
+	if (_massPrior) {
+#ifdef _OPENMP
+#pragma omp critical(massPrior)
+#endif
+		{
+			phaseSpace *= _massPrior->Eval(parameters[3*_nPart-6]);
+		}
+	}
+
+	timerTot.Stop();
+#ifdef _OPENMP
+#pragma omp atomic
+#endif
+	_funcCallInfo[LOGAPRIORIPROBABILITY].totalTime += timerTot.RealTime();
+
+	return std::log(phaseSpace);
+}
+
+
+double
+rpwa::importanceSampler::LogLikelihood(const std::vector<double>& parameters)
+{
+#ifdef _OPENMP
+#pragma omp atomic
+#endif
+	++(_funcCallInfo[LOGLIKELIHOOD].nmbCalls);
+	TStopwatch timerTot;
+	timerTot.Start();
+
+	if (_phaseSpaceOnly) {
+		return -1;
```

Comment:
its as good a choice as any other finite (and constant within one BAT run) number, nonetheless I changed it to 0